projects/saturn/src/main/java/io/fairspace/saturn/webdav/DavFactory.java

```diff
@@ -136,18 +143,19 @@ public Access getAccess(org.apache.jena.rdf.model.Resource subject) {
             access = min(access, Access.List);
         }
 
-        if(access == Access.None && userService.currentUser().isAdmin()) {
+        if (access == Access.None && userService.currentUser().isAdmin()) {
             return Access.List;
         }
 
         return access;
     }
 
     private Access getExtraStorageAccess(org.apache.jena.rdf.model.Resource subject) {
-        if (subject.hasProperty(FS.createdBy, currentUserResource()) && subject.hasProperty(RDF.type, FS.File)) {
-            return Access.Manage;
+        if (subject.equals(this.rootSubject)) {
```

Comment:
could the order of if statements have a result that you get read access, while you should have write access?

Reply:
yes, the order matters, this is to protect the root folder being deleted by a user (it is created on application startup and should not be deleted), so it has to be read-only.